Tools/export.sh

```diff
@@ -33,14 +34,14 @@ fi
 export PYTHON=${PYTHON:=$(which python3)}
 
 # Esp8266
-export ESP_HOME=${ESP_HOME:=/opt/esp-quick-toolchain}
+export ESP_HOME=${ESP_HOME:=${_SMIG_BASE}/esp-quick-toolchain}
 
 # Esp32
-export IDF_PATH=${IDF_PATH:=/opt/esp-idf}
-export IDF_TOOLS_PATH=${IDF_TOOLS_PATH:=/opt/esp32}
+export IDF_PATH=${IDF_PATH:=${_SMIG_BASE}/esp-idf}
+export IDF_TOOLS_PATH=${IDF_TOOLS_PATH:=${_SMIG_BASE}/esp32}
 
 # Rp2040
-export PICO_TOOLCHAIN_PATH=${PICO_TOOLCHAIN_PATH:=/opt/rp2040}
+export PICO_TOOLCHAIN_PATH=${PICO_TOOLCHAIN_PATH:=${_SMIG_BASE}/rp2040}
 
 # Provide non-apple CLANG (e.g. for rbpf library)
 if [ -n "$GITHUB_ACTIONS" ] && [ "$(uname)" = "Darwin" ]; then
```

Comment:
No. Using the `Sming` install location is worse than using `/opt`. Whilst there are permissions issues with `/opt` I do believe it is the most sensible default. See https://www.baeldung.com/linux/opt-directory and https://en.wikipedia.org/wiki/Filesystem_Hierarchy_Standard.

Incidentally I never install `Sming` in /opt but use sandboxes on a separate ZFS volume. I specifically *never* put bloatey tool installations (or the IDF) there. Thus the locations of toolchains should be decoupled from the location for Sming.

The procedure for customising install directories is outlined here https://sming.readthedocs.io/en/stable/getting-started/config.html. That should meet your needs. In particular, the ESP32 stuff has separate `IDF_PATH` for the framework (which needs to be writeable) and `IDF_TOOLS_PATH` for the toolchains.

Reply:
I reverted these changes. 

I handle the non-standart tools installation using `direnv` now.

With Sming checked out to my home directory, and adding a small `.envrc` to this directory (and every directory with projects using Sming):
```
layout Sming
```
and adding
```
layout_Sming() {

    _SMIG_BASE=${HOME}/Sming
    _SMIG_TOOLS=${_SMIG_BASE}/.tools
    export ESP_HOME=${_SMIG_TOOLS}/esp-quick-toolchain
    export IDF_PATH=${_SMIG_TOOLS}/esp-idf
    export IDF_TOOLS_PATH=${_SMIG_TOOLS}/esp32
    export PICO_TOOLCHAIN_PATH=${_SMIG_TOOLS}/rp2040

    . ${_SMIG_BASE}/Tools/export.sh
}
```
to `~/.config/direnv/direnvrc` which handles the required path settings.